test/widgets/actions_test.dart

```diff
@@ -101,18 +101,129 @@ void main() {
           });
     });
 
+    testWidgets('CombinedFeedNarrow on legacy server', (WidgetTester tester) async {
```

Comment:
When moving code around while refactoring, while there could be some cases where a split is not possible, it is generally preferred to have separate commits for the move and for the refactor to make it easier to review.